u2f-ref-code/java/src/com/google/u2f/codec/RawMessageCodec.java

```diff
@@ -95,10 +91,14 @@ public static RegisterResponse decodeRegisterResponse(byte[] data) throws U2FExc
       inputStream.readFully(userPublicKey);
       byte[] keyHandle = new byte[inputStream.readUnsignedByte()];
       inputStream.readFully(keyHandle);
-      X509Certificate attestationCertificate = (X509Certificate) CertificateFactory.getInstance(
-          "X.509").generateCertificate(inputStream);
-      byte[] signature = new byte[inputStream.available()];
-      inputStream.readFully(signature);
+
+      // We don't know the length of the signature or cert chain, so we tread carefully
+      byte[] certsAndSig = new byte[inputStream.available()];
+      inputStream.readFully(certsAndSig);
+
+      X509Certificate[] attestationCertificateChain = X509Util.parseCertificateChain(certsAndSig);
+      byte[] signature = Arrays.copyOfRange(certsAndSig,
```

Comment:
This took some reading to figure out. I think what you're doing is:
1. Assuming that the signature is all the bytes that remain at the end of the certificate chain.
2. Decoding, then re-encoding, the certificate chain, to figure out where the end of the encoded certificate chain is.

Is that right? Maybe a comment to that effect would help?